source.go

```diff
@@ -107,8 +109,25 @@ func (a *sourcePluginAdapter) Configure(ctx context.Context, req cpluginv1.Sourc
 }
 
 func (a *sourcePluginAdapter) Start(ctx context.Context, req cpluginv1.SourceStartRequest) (cpluginv1.SourceStartResponse, error) {
-	ctx, a.openCancel = context.WithCancel(ctx)
-	err := a.impl.Open(ctx, req.Position)
+	// create a new context, so we can control when it's canceled
+	ctxOpen := context.Background()
+	ctxOpen, a.openCancel = context.WithCancel(ctxOpen)
```

Comment:
I'm wondering, why don't we simply do `ctxOpen, a.openCancel = context.WithCancel(context.Background())`?

Or maybe even `ctxOpen, a.openCancel = context.WithCancel(ctx)`?

Reply:
First option won't work unless you declare `ctxOpen` a line before with `var ctxOpen context.Context`, not really an improvement IMO.

Second option is essentially what we did before but it had a flaw - the context could be cancelled even after `Open` returns (and it actually happens), so that context is then not suitable for spawning long-running goroutines.

Reply:
Thanks! TIL a bit more about multiple short var declarations.: )